libenv/sysinfo.c

```diff
@@ -3335,6 +3335,76 @@ void GetDefVars(EvalContext *ctx)
         "jq --compact-output --monochrome-output --ascii-output --unbuffered --sort-keys",
         CF_DATA_TYPE_STRING, "invocation,source=agent,command_name=jq");
 }
+
+/*******************************************************************/
+
+static void SysOSNameHuman(EvalContext *ctx)
+{
+    assert(ctx != NULL);
+    const char *const lval = "os_name_human";
+    const char *const tags = "source=agent";
+
+    /**
+     * Order is important. The first if-statement is more specific and wins.
+     * E.g. prefer Ubuntu over Debian.
+     */
+    if (EvalContextClassGet(ctx, NULL, "ubuntu") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Ubuntu", CF_DATA_TYPE_STRING, tags);
```

Comment:
I like to add tags for exactly where the variable came from, so in this case:

```suggestion
        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
                                      "Ubuntu", CF_DATA_TYPE_STRING, "source=agent,derived-from=ubuntu");
```

Like I've done here: https://github.com/cfengine/masterfiles/blob/8748fbd14323e18a326ae7a2ddc0a7ca2fda1bfd/inventory/os.cf#L33-L35

(And same for all cases below).

@nickanderson do you agree?

Reply:
It also makes it very easy to grep for where the variable was defined, just grep for the tags.